sdk/src/Services/EC2/Custom/_bcl/Util/EC2Metadata.cs

```diff
@@ -392,9 +392,12 @@ private static List<string> GetItems(string path, int tries, bool slurp)
                 else
                     request = WebRequest.Create(EC2_METADATA_ROOT + path) as HttpWebRequest;
 
-                request.Timeout = (int)TimeSpan.FromSeconds(5).TotalMilliseconds;
+                var responseTask = request.GetResponseAsync();
 
-                using (var response = request.GetResponse())
+                if (!responseTask.Wait((int)TimeSpan.FromSeconds(5).TotalMilliseconds))
+                    throw new WebException("Timedout");
```

Comment:
After taking another look at this, I am not aborting the task on timeout. That cancellation is not trivial since it is not possible to pass in a cancellation token.